analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RedundantArgument.CSharp12.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+
+class DefaultLambdaParameters
+{
+    void SingleDefaultArgument()
+    {
+        var f = (int i = 42) => i;
+        f();       // Compliant
+        f(40);     // Compliant
+        f(42);     // Noncompliant
+        f(41 + 1); // Noncompliant, expression results into the default value
+    }
+
+    void MultipleDefaultArguments()
+    {
+        var f = (int i = 41, int j = 42) => i + j;
+        f();       // Compliant
+        f(42);     // Compliant
+        f(42, 42); // Noncompliant
+        f(41, 42); // Multiple violations
+        //^^
+        //    ^^@-1
+    }
+
+    void NamedArguments()
+    {
+        var f = (int i = 41, int j = 42, int z = 43) => i;
+        f(i: 42); // Error CS1746  The delegate '<anonymous delegate>' does not have a parameter named 'i'
+    }
+}
+
+// https://github.com/SonarSource/sonar-dotnet/issues/8096
+namespace Repro_8096
+{
+    class PrimaryConstructors
+    {
+        void SingleDefaultArgument()
+        {
+            _ = new C1();       // Compliant
+            _ = new C1(41);     // Compliant
+            _ = new C1(42);     // FN
+            _ = new C1(41 + 1); // FN, expression results into the default value
+        }
+
+        void MultipleDefaultArguments()
+        {
+            _ = new C1();       // Compliant
+            _ = new C2(42);     // Compliant
+            _ = new C2(42, 42); // FN
+            _ = new C2(41, 42); // FN, multiple violations
+                                //         ~~
```

Comment:
Educational: what is this `~~` doing?

Reply:
Very interesting, I never encountered such a method to indicate the FN location. Not a strong opinion: I think we should decide whether we want to keep it and document it somewhere (as of now it's not mentioned in the test cases [comment pattern](https://github.com/SonarSource/sonar-dotnet/blob/d56b3718f2dc350e9427dbc51af9d514a8e6c7c6/analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/IssueLocationCollector.cs#L34-L91)), or avoid using it entirely. WDYT?

Reply:
It's not defined in the `IssueLocationCollector` because it's an FN location, so it cannot be asserted.
Moreover, it keeps being vertically aligned by Visual Studio to the `// FN, multiple violations` of the line above.
So I have removed it.